forgerock-auth/src/main/java/org/forgerock/android/auth/AppAuthFragment2.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+package org.forgerock.android.auth
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.annotation.RestrictTo
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+import androidx.fragment.app.FragmentManager
+import kotlinx.coroutines.flow.MutableStateFlow
+import net.openid.appauth.AuthorizationResponse
+import org.forgerock.android.auth.centralize.BrowserLauncher
+import org.forgerock.android.auth.centralize.Launcher
+
+/**
+ * Headless Fragment to receive callback result from AppAuth library
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY)
```

Comment:
whats the issue with other approach ? i understand we sont need a extra call..but do you notice any issues on the extension approach ?

Reply:
Yes, the issue about "Don't keep activities" with consent. The issue is when finishing up browser authentication, the browser redirects back to the App, and the App calls onCreate, and resumes the previous state, when resumes previous state, it calls the launch again, which launches the browser again, which makes it never back to the App again.